torchao/quantization/prototype/qat/api.py

```diff
@@ -4,11 +4,75 @@
 # This source code is licensed under the license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Any, List
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, List, Optional
 
 import torch
 
 from torchao.quantization.unified import TwoStepQuantizer
+from torchao.quantization.quant_primitives import ZeroPointDomain
+
+
+class QuantizationGranularity(Enum):
+    PER_CHANNEL = "per_channel"
+    PER_TOKEN = "per_token"
+    PER_GROUP = "per_group"
+
+
+@dataclass
+class FakeQuantizeConfig:
+    """
+    Config for how to fake quantize weights or activations.
+
+    args:
+        bit_width: number of bits to simulate during fake quantization
+        granularity: granularity of scales and zero points, one of:
+            'per_token', 'per_channel', or 'per_group'
+        group_size: size of each group for 'per_group' granularity
+        symmetric: whether to use symmetric (default) or asymmetric quantization
+        scale_precision: scale dtype (default torch.fp32)
+        zero_point_precision: zero point dtype (default torch.int32)
+        zero_point_domain: whether zero point is in integer (default) or float domain
+        dynamic: whether to use dynamic (defualt) or static scale and zero points
+        range_learning: whether to learn scale and zero points during training (coming soon)
+    """
+    bit_width: int
+    granularity: Optional[QuantizationGranularity] = None
+    group_size: Optional[int] = None
```

Comment:
can be reusing https://github.com/pytorch/ao/blob/dec03131484b5717e9fb8b6c0c25f41ec30e9fef/torchao/quantization/observer.py#L20 here and also we plan to rename it to `Granularity` for better clarity